### ISSUE TITLE
Remove unused Functions

### ISSUE THREAD

**Issue by CasparKielwein** [scp-fs2open/fs2open.github.com#1955]
FSO contains a large number of unused functions and variable.

Over time the amount of unused code has grown so large, that compiler warnings needed to be disabled.
This situation is obviously unsustainable.

I've gone through the codebase and removed a lot of obviously unused code and assembled a few PRs.
My workflow has been:
* Run clang tidy to check for unused functions.
* For each function it warns, grep in entire codebase
* If there are no results except definition and/or implementation -> remove function.
* The same for some obvious global variables and defines.
* Ignore scripting related code as the lua bindings make checking for unused functions non-trivial
* ignore library code like lib/ ,vector math, 3dmath.

Since large changes like these are usually hard to review, I suggest the following workflow:
Reviewers have a look and if they are unsure about a particular function or file, I remove it from the PR in an effort to have the noncontroversial changes merged as fast as possible.
I will then later make another PR with the more controversial changes to discuss them.

**Comment by Goober5000** [comment]
`cmeasure_select_next` was an important part of WMC's countermeasure overhaul - the ability to select multiple types of countermeasures.  Unfortunately WMC hasn't been around for ages.

**Comment by CasparKielwein** [comment]
We could tag the last commit before it was removed, in case someone wants to pick up where he left off.

After #1958 is closed, there is one warning left:
`qtfred/src/mission/FredRenderer.cpp:156:5: warning: ‘int {anonymous}::get_subsys_bounding_rect(object*, ship_subsys*, int*, int*, int*, int*)’ defined but not used`

I haven't touched qtfred.
